Mathlib/Tactic/NormNum/Core.lean

```diff
@@ -215,55 +216,6 @@ mutual
     (·.1) <$> Simp.main e ctx (methods := methods)
 end
 
--- FIXME: had to inline a bunch of stuff from `simpGoal` here
-/--
-The core of `norm_num` as a tactic in `MetaM`.
-
-* `g`: The goal to simplify
-* `ctx`: The simp context, constructed by `mkSimpContext` and
-  containing any additional simp rules we want to use
-* `fvarIdsToSimp`: The selected set of hypotheses used in the location argument
-* `simplifyTarget`: true if the target is selected in the location argument
-* `useSimp`: true if we used `norm_num` instead of `norm_num1`
--/
-def normNumAt (g : MVarId) (ctx : Simp.Context) (fvarIdsToSimp : Array FVarId)
-    (simplifyTarget := true) (useSimp := true) :
-    MetaM (Option (Array FVarId × MVarId)) := g.withContext do
-  g.checkNotAssigned `norm_num
-  let mut g := g
-  let mut toAssert := #[]
-  let mut replaced := #[]
-  for fvarId in fvarIdsToSimp do
-    let localDecl ← fvarId.getDecl
-    let type ← instantiateMVars localDecl.type
-    let ctx := ctx.setSimpTheorems (ctx.simpTheorems.eraseTheorem (.fvar localDecl.fvarId))
-    let r ← deriveSimp ctx useSimp type
-    match r.proof? with
-    | some _ =>
-      let some (value, type) ← applySimpResult g (mkFVar fvarId) type r
-        | return none
-      toAssert := toAssert.push { userName := localDecl.userName, type, value }
-    | none =>
-      if r.expr.isConstOf ``False then
```

Comment:
It seems this logic is getting lost.

Reply:
This is handled by the flag `(mayCloseGoalFromHyp := true)`, or am I missing your point?

Reply:
No, you're right!